src/worker/tasks/mineTransactionWorker.ts

```diff
@@ -160,6 +173,32 @@ const _mineTransaction = async (
         resendCount: resendCount + 1,
       });
     }
+  } else {
+    const erroredTransaction: ErroredTransaction = {
+      ...sentTransaction,
+      status: "errored",
+      errorMessage: "Transaction timed out.",
+    };
+    job.log(`Transaction timed out: ${stringify(erroredTransaction)}`);
+
+    await TransactionDB.set(erroredTransaction);
+    await enqueueTransactionWebhook(erroredTransaction);
+    _reportUsageError(erroredTransaction);
+
+    // Release the nonce to allow it to be reused or cancelled.
+    job.log(`Recycling nonce: ${sentTransaction.nonce}`);
+
+    await removeSentNonce(
+      sentTransaction.chainId,
+      sentTransaction.from,
+      sentTransaction.nonce,
+    );
+
+    await recycleNonce(
+      sentTransaction.chainId,
+      sentTransaction.from,
+      sentTransaction.nonce,
+    );
```

Comment:
This shouldn't be here. All this logic already exists on L233 of this file in the "failed" listener after the last attempt. Effectively when a job tries the max num of attempts, it is considered "timed out" and:
- errors the tx
- sends the error webhook/usage event
- recycles the nonce
You just need to add the `removeSentNonce` step there.

Reply:
thanks, good catch. wonder how a test could have been setup to catch this, since it's very criticial